Protobuf lite fails to link when building with shared libs

<!--

NOTE: this form is for bug reports only. For questions or troubleshooting, please see the protobuf mailing list: https://groups.google.com/forum/#!forum/protobuf

-->

**What version of protobuf and what language are you using?**
Version: v3.17.3
Language: C++

**What operating system (Linux, Windows, ...) and version?**
Windows 10

**What runtime / compiler are you using (e.g., python version or gcc version)**
Visual Studio 2019 (16.10.2)

**What did you do?**
Steps to reproduce the behavior:
1. Run the VS Developer Command Prompt (64-bit)
2. 
```
mkdir cmake_output
cd cmake_output
cmake .\protobuf-3.17.3\cmake -Dprotobuf_BUILD_SHARED_LIBS=ON
```
3. Build the generated `libprotobuf-lite` project from the `cmake_output` folder through the VS IDE or msbuild command line. It doesn't appear that Debug or Release configuration matters.

**What did you expect to see**
Successful build.

**What did you see instead?**
Linker error with missing definitions:
```
2>field_access_listener.obj : error LNK2019: unresolved external symbol "private: void __cdecl google::protobuf::internal::LazyDescriptor::Once(void)" (?Once@LazyDescriptor@internal@protobuf@google@@AEAAXXZ) referenced in function "public: class google::protobuf::Descriptor const * __cdecl google::protobuf::internal::LazyDescriptor::Get(void)" (?Get@LazyDescriptor@internal@protobuf@google@@QEAAPEBVDescriptor@34@XZ)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: struct google::protobuf::Descriptor::ExtensionRange const * __cdecl google::protobuf::Descriptor::FindExtensionRangeContainingNumber(int)const " (?FindExtensionRangeContainingNumber@Descriptor@protobuf@google@@QEBAPEBUExtensionRange@123@H@Z) referenced in function "public: bool __cdecl google::protobuf::Descriptor::IsExtensionNumber(int)const " (?IsExtensionNumber@Descriptor@protobuf@google@@QEBA_NH@Z)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: struct google::protobuf::Descriptor::ReservedRange const * __cdecl google::protobuf::Descriptor::FindReservedRangeContainingNumber(int)const " (?FindReservedRangeContainingNumber@Descriptor@protobuf@google@@QEBAPEBUReservedRange@123@H@Z) referenced in function "public: bool __cdecl google::protobuf::Descriptor::IsReservedNumber(int)const " (?IsReservedNumber@Descriptor@protobuf@google@@QEBA_NH@Z)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "private: bool __cdecl google::protobuf::FieldDescriptor::is_map_message_type(void)const " (?is_map_message_type@FieldDescriptor@protobuf@google@@AEBA_NXZ) referenced in function "public: bool __cdecl google::protobuf::FieldDescriptor::is_map(void)const " (?is_map@FieldDescriptor@protobuf@google@@QEBA_NXZ)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "private: static void __cdecl google::protobuf::FieldDescriptor::TypeOnceInit(class google::protobuf::FieldDescriptor const *)" (?TypeOnceInit@FieldDescriptor@protobuf@google@@CAXPEBV123@@Z) referenced in function "public: enum google::protobuf::FieldDescriptor::Type __cdecl google::protobuf::FieldDescriptor::type(void)const " (?type@FieldDescriptor@protobuf@google@@QEBA?AW4Type@123@XZ)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: struct google::protobuf::EnumDescriptor::ReservedRange const * __cdecl google::protobuf::EnumDescriptor::FindReservedRangeContainingNumber(int)const " (?FindReservedRangeContainingNumber@EnumDescriptor@protobuf@google@@QEBAPEBUReservedRange@123@H@Z) referenced in function "public: bool __cdecl google::protobuf::EnumDescriptor::IsReservedNumber(int)const " (?IsReservedNumber@EnumDescriptor@protobuf@google@@QEBA_NH@Z)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: class google::protobuf::FileDescriptor const * __cdecl google::protobuf::FileDescriptor::dependency(int)const " (?dependency@FileDescriptor@protobuf@google@@QEBAPEBV123@H@Z) referenced in function "public: class google::protobuf::FileDescriptor const * __cdecl google::protobuf::FileDescriptor::public_dependency(int)const " (?public_dependency@FileDescriptor@protobuf@google@@QEBAPEBV123@H@Z)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: virtual __cdecl google::protobuf::DescriptorPool::ErrorCollector::~ErrorCollector(void)" (??1ErrorCollector@DescriptorPool@protobuf@google@@UEAA@XZ) referenced in function "public: virtual void * __cdecl google::protobuf::DescriptorPool::ErrorCollector::`vector deleting destructor'(unsigned int)" (??_EErrorCollector@DescriptorPool@protobuf@google@@UEAAPEAXI@Z)
2>field_access_listener.obj : error LNK2019: unresolved external symbol "public: void __cdecl google::protobuf::DescriptorPool::InternalDontEnforceDependencies(void)" (?InternalDontEnforceDependencies@DescriptorPool@protobuf@google@@QEAAXXZ) referenced in function "public: void __cdecl google::protobuf::DescriptorPool::InternalSetLazilyBuildDependencies(void)" (?InternalSetLazilyBuildDependencies@DescriptorPool@protobuf@google@@QEAAXXZ)
2>field_access_listener.obj : error LNK2001: unresolved external symbol "private: static enum google::protobuf::FieldDescriptor::CppType const * const google::protobuf::FieldDescriptor::kTypeToCppTypeMap" (?kTypeToCppTypeMap@FieldDescriptor@protobuf@google@@0QBW4CppType@123@B)
2>field_access_listener.obj : error LNK2001: unresolved external symbol "private: static char const * const * const google::protobuf::FieldDescriptor::kTypeToName" (?kTypeToName@FieldDescriptor@protobuf@google@@0QBQEBDB)
2>field_access_listener.obj : error LNK2001: unresolved external symbol "private: static char const * const * const google::protobuf::FieldDescriptor::kCppTypeToName" (?kCppTypeToName@FieldDescriptor@protobuf@google@@0QBQEBDB)
```

**Anything else we should know about your project / environment**
Standard environment.

--- a/cmake/libprotobuf-lite.cmake
+++ b/cmake/libprotobuf-lite.cmake
@@ -3,7 +3,6 @@ set(libprotobuf_lite_files
   ${protobuf_source_dir}/src/google/protobuf/arena.cc
   ${protobuf_source_dir}/src/google/protobuf/arenastring.cc
   ${protobuf_source_dir}/src/google/protobuf/extension_set.cc
-  ${protobuf_source_dir}/src/google/protobuf/field_access_listener.cc
   ${protobuf_source_dir}/src/google/protobuf/generated_enum_util.cc
   ${protobuf_source_dir}/src/google/protobuf/generated_message_table_driven_lite.cc
   ${protobuf_source_dir}/src/google/protobuf/generated_message_util.cc

--- a/cmake/libprotobuf.cmake
+++ b/cmake/libprotobuf.cmake
@@ -11,6 +11,7 @@ set(libprotobuf_files
   ${protobuf_source_dir}/src/google/protobuf/dynamic_message.cc
   ${protobuf_source_dir}/src/google/protobuf/empty.pb.cc
   ${protobuf_source_dir}/src/google/protobuf/extension_set_heavy.cc
+  ${protobuf_source_dir}/src/google/protobuf/field_access_listener.cc
   ${protobuf_source_dir}/src/google/protobuf/field_mask.pb.cc
   ${protobuf_source_dir}/src/google/protobuf/generated_message_reflection.cc
   ${protobuf_source_dir}/src/google/protobuf/generated_message_table_driven.cc

